lib/utils/grpc.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"context"
+
+	"github.com/gravitational/trace/trail"
+	"google.golang.org/grpc"
+)
+
+func ErrorConvertUnaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	resp, err := handler(ctx, req)
+	return resp, trail.ToGRPC(err)
+}
+
+func ErrorConvertStreamInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	return trail.ToGRPC(handler(srv, ss))
+}
+
+// ChainUnaryServerInterceptors takes 1 or more grpc.UnaryServerInterceptors and
+// chains them into a single grpc.UnaryServerInterceptor. The first interceptor
+// will be the outer most, while the last interceptor will be the inner most
+// wrapper around the real call.
+func ChainUnaryServerInterceptors(first grpc.UnaryServerInterceptor, rest ...grpc.UnaryServerInterceptor) grpc.UnaryServerInterceptor {
```

Comment:
Doesn't grpc_middleware does this? Could we import it instead?

Ditto for stream chaining.

https://pkg.go.dev/github.com/grpc-ecosystem/go-grpc-middleware#pkg-index

Reply:
Looks like that library does implement the same thing, it's pretty common. But I've already written the few lines of code it takes, is it really worth adding a dependency instead?

Reply:
I don't mind pulling from grpc-ecosystem, it's a common import. My next ask, if we keep your code, is to write a few tests for it. Your call.

Reply:
Just saw the tests, btw. :)